front/src/components/oauth2/actions.js

```diff
@@ -0,0 +1,112 @@
+import { RequestStatus } from '../../utils/consts';
+
+const actions = store => ({
+  updateIntegrationName(state, e) {
+    store.setState({
+      integrationName: e
+    });
+  },
+  updateClientId(state, e) {
+    store.setState({
+      clientId: e.target.value
+    });
+  },
+  updateSecretId(state, e) {
+    store.setState({
+      secretId: e.target.value
+    });
+  },
+  async getCurrentConfig(state) {
+    store.setState({
+      oauth2GetStatus: RequestStatus.Getting
+    });
+
+    const returnServiceId = await state.httpClient.get(`/api/v1/service/${state.integrationName}/getServiceId`);
+
+    const returnGetConfig = await state.httpClient.get('/api/v1/service/oauth2/client/current-config', {
+      integrationName: state.integrationName,
+      serviceId: returnServiceId.result.serviceId
+    });
+
+    if (returnGetConfig) {
+      // Case of config found
+      store.setState({
+        clientIdInDb: returnGetConfig.clientId,
+        oauth2ErrorMsg: null,
+        oauth2GetStatus: RequestStatus.Success
+      });
+    } else {
+      store.setState({
+        oauth2ErrorMsg: null,
+        oauth2GetStatus: RequestStatus.Success
+      });
+    }
+  },
+  async startConnect(state) {
+    store.setState({
+      oauth2GetStatus: RequestStatus.Getting
+    });
+    try {
+      if (!state.secretId || !state.clientId) {
+        store.setState({
+          oauth2GetStatus: RequestStatus.Error,
+          oauth2ErrorMsg: 'errorEmptyConfig'
+        });
+        return;
+      }
+
+      const returnSaveVar = await state.httpClient.post(`/api/v1/service/${state.integrationName}/savevar`, {
+        clientId: state.clientId,
+        secretId: state.secretId,
+        integrationName: state.integrationName
+      });
+
+      let serviceId;
+      if (returnSaveVar.success === true) {
+        serviceId = returnSaveVar.result.serviceId;
+      } else {
+        store.setState({
+          oauth2GetStatus: RequestStatus.Error,
+          oauth2ErrorMsg: 'errorServiceId'
+        });
+      }
+
+      const returnValue = await state.httpClient.post('/api/v1/service/oauth2/client/authorization-uri', {
+        clientId: state.clientId,
+        secretId: state.secretId,
+        integrationName: state.integrationName,
+        serviceId
+      });
+      if (returnValue.success === true) {
+        window.location = returnValue.authorizationUri;
+      } else {
+        store.setState({
+          oauth2GetStatus: RequestStatus.Error,
+          oauth2ErrorMsg: 'errorAuthorizationUri'
+        });
+        return;
+      }
+    } catch (e) {
+      store.setState({
+        oauth2GetStatus: RequestStatus.Error,
+        oauth2ErrorMsg: 'errorAuthorizationUri'
+      });
+      return;
+    }
+  },
+  async unConnect(state) {
+    store.setState({
+      oauth2GetStatus: RequestStatus.Getting
+    });
+
+    await state.httpClient.get(`/api/v1/service/${state.integrationName}/deleteConfig`);
```

Comment:
I didn't find any implementation of this.
You should use your "oath2 controller".

Reply:
for me this call must delete all element of specific integration (to delete all element in database linked to integration like devices,varibles, ...), it's why I have not implementation in oauth2controler